domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputContainsRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,189 @@
+package org.oppia.domain.classify.rules.textinput
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.model.InteractionObject
+import org.robolectric.annotation.Config
+import java.lang.IllegalStateException
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [TextInputContainsRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class TextInputContainsRuleClassifierProviderTest {
+  private val NON_NEGATIVE_VALUE = createNonNegativeInt(value = 1)
+  private val STRING_ANSWER = createString(value = "this is a test")
+  private val STRING_INPUT_BEGINNING = createString(value = "this is")
+  private val STRING_INPUT_MIDDLE = createString(value = "is a")
+  private val STRING_INPUT_END = createString(value = "a test")
+  private val STRING_INPUT_EXTENDS_ANSWER = createString(value = "this is a test i will break")
+  private val STRING_INPUT_NO_SPACES = createString(value = "thisisatest")
+  private val STRING_INPUT_EXTRA_SPACES = createString(value = " this   is  a  test ")
+  private val STRING_INPUT_NOT_ANSWER = createString(value = "an answer")
+
+  @Inject
+  internal lateinit var textInputContainsRuleClassifierProvider:
+    TextInputContainsRuleClassifierProvider
+
+  private val inputContainsRuleClassifier by lazy {
+    textInputContainsRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testStringAnswer_stringInput_sameString_bothValuesMatch() {
+    val inputs = mapOf("x" to STRING_ANSWER)
+
+    val matches = inputContainsRuleClassifier.matches(answer = STRING_ANSWER, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testStringAnswer_stringInput_answerContainsInputAtBeginning_bothValuesMatch() {
+    val inputs = mapOf("x" to STRING_INPUT_BEGINNING)
```

Comment:
Hmm, since contains is heavily nuanced on the exact value of the inputs it may be worth just inlining all of the constants for these tests. Without them, it's hard to verify the test is testing what it's supposed to be without looking outside of the test body.

Reply:
OK, will do!

Reply:
Should I inline the constant I named "STRING_ANSWER"? I see why it could be confusing and require those without context to look outside the test to understand what is going on, but at the same time I use that string in almost every test - isn't it good practice to have it be a constant when you use it so much? And if I do keep it a constant, maybe re-naming it to something like "STRING_THIS_IS_A_TEST" or something like that will help?